llama/llama.go

```diff
@@ -637,8 +637,8 @@ func SchemaToGrammar(schema []byte) []byte {
 	cStr := C.CString(string(schema))
 	defer C.free(unsafe.Pointer(cStr))
 
-	// Allocate buffer for grammar output with reasonable size
-	const maxLen = 32768 // 32KB
+	// Allocate buffer for grammar based on schema length but with upper bound
+	maxLen := min(1024*1024*1024, len(schema)*4)
```

Comment:
Thank for fixing this @rick-github !!

Wouldn't this mean an upper bound of 1GB? Should we lower that a bit?

Reply:
perfect!